### PR TITLE
HELP SEND HELP

### PR DESCRIPTION
**Pull request by slin99**
I was forced to write an trapdoor writeup

### CODE REVIEW

**Review by intrigus-lgtm — CHANGES_REQUESTED** [review]
post.html change bad

File: _posts/2024-06-01-galois.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: "Trapdoor author writeup"
```
Comment (rad4day) [inline _posts/2024-06-01-galois.md:3]: A common `GPNCTF:` prefix or category would be nice

File: _posts/2024-06-01-galois.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+But this essentially means we can combine multiple instances to get an  iterator that skips more elements. This construction is bascially a kind of chinese remainder theorem.
+My  solve script especially the last part is extremly ugly so I wont publish it, but I have included the challenge files somewhere so feel free to try to solve it yourself.
+
+### Related resources 
+If you want to know more about Galois theory, consider visiting a university course if you have the chance, otherwise there are many great online resources and books, such as [this](https://www.maths.ed.ac.uk/~tl/gt/gt.pdf) (be warned that most of Galois theory is not about finite fields), but you will probably want to start with an introduction to groups and algebra in general first.
+Somewhere here should be a link to the challenge files
+[Here](https://files.ctf.kitctf.de/trapdoor/a053f0f33932977546aa3e9720188e42cd8a1c6921ce95908f5f35766e2f53d6/trapdoor.tar.gz) the handout
```
Comment (rad4day) [inline _posts/2024-06-01-galois.md:60]: This url will potentially be offline soonish™

File: repo.bundle
Comment (rad4day) [inline repo.bundle:1]: What's that? Why is it here?

File: _posts/2024-06-01-galois.md (hunk 2 above)
Comment (rad4day) [reply _posts/2024-06-01-galois.md:60]: @intrigus-lgtm we should think about a solution for this when we archive the event page.

**Review by rad4day — CHANGES_REQUESTED** [review]

File: _posts/2024-06-01-galois.md (hunk 2 above)
Comment (intrigus-lgtm) [reply _posts/2024-06-01-galois.md:60]: Handout files should be added to the static github pages repository.
Only question would be what to do with sourceless challenges.
Without a handout and without a live instance they are unplayable :thinking: 
I think last year all challenges had a handout with the source (if we ignore the github actions challenge, which however has been preserved in another way), but this year that's not true.

File: _posts/2024-06-01-galois.md (hunk 1 above)
Comment (slin99) [reply _posts/2024-06-01-galois.md]: done

File: _posts/2024-06-01-galois.md (hunk 2 above)
Comment (slin99) [reply _posts/2024-06-01-galois.md:60]: not my beer

File: repo.bundle
Comment (slin99) [reply repo.bundle:1]: artifact of first try of doing pull request via dm . removed

**Comment by slin99** [comment]
changes in post.html are important

File: _posts/2024-04-08-rsa-paddings.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "Swampctf coppercrypto and bad paddings in RSA schemes"
+categories: learning
+author: s1nn105 
+mathjax: true
+---
+A little contribution to revive the KITCTF blog.
```
Comment (intrigus-lgtm) [inline _posts/2024-04-08-rsa-paddings.md:8]: Remove that sentence.

File: _posts/2024-04-08-rsa-paddings.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+c = pow(m,e,n)
+
+with open('out.txt', 'w') as fout:
+	fout.write(f'n = {n}\n')
+	fout.write(f'e = {e}\n')
+	fout.write(f'c = {c}\n')
+```
+The outputs (out.txt) were also given.
+There are serveral things wrong with this code. 
+First an foremost the fact that it does not use SageMath.
```
Comment (intrigus-lgtm) [inline _posts/2024-04-08-rsa-paddings.md:39]:
```suggestion
First and foremost the fact that it does not use SageMath.
```

File: _posts/2024-04-08-rsa-paddings.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Jokes aside  the weakness lays in the padding.
+The script loads the flag, pads it with null bytes and then encrypts it 
+with an randomly generated RSA key. for the RSA key we use two 512 bit primes. This 1024 bit modulus is accompanied by 3 as the public exponent.
+This small public exponent is dubious at best. While a small public exponent in itself might not be a problem, it is for sure not optimal an will enable multiple attacks on this. 
+
+
+This challenge essentially calls for the coppersmith  theorem, but we shall not bring guns to a knife fight. [At least not yet.]
+
+Padding with null bytes is unusal, and that for a good reason:
```
Comment (intrigus-lgtm) [inline _posts/2024-04-08-rsa-paddings.md:48]:
```suggestion
Padding with null bytes is unusual, and that for a good reason: 
```

File: _posts/2024-04-08-rsa-paddings.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Lets step back for one second. Assume we know the length of the flag and thus the length of the padding, the padding becomes no more than a constant $$pad$$. Encrypting the padded flag $$ \textit{Enc}_\textit{padded}(pk,pad) = (flag \cdot pad)^3 \textit{ mod } n  = C $$. 
+
+However since we  know the padding $$pad$$ we can (with a little bit of luck) compute $$pad^{-1} \textit{ mod } n$$ using the known public key we can cumpute $$C \cdot (pad^{-1})^3 \textit{ mod } n$$.
+Simplifying this results in $$ = flag^3 \textit{ mod } n =: C_f $$.
+
+
+Hitherto we did not use the low public exponent, but now we can assume (due to the "large" public modulus and the small public exponent) that the ciphertext would not be much larger if it would be calculated in the integers instead of the modular arithmetic. Using this assumption we can simply recover the flag by computing the third root of $$C_f$$. Implementing this is trivial, you can try this your self, but seems a bit cheap if you consider that this "trick" does only work in certain circumstances. A more general approach will be presented in the next chapter.
```
Comment (intrigus-lgtm) [inline _posts/2024-04-08-rsa-paddings.md:63]: What next chapter???

File: _posts/2024-06-01.md
Comment (intrigus-lgtm) [inline _posts/2024-06-01.md:1]: Duplicate?

**Review by intrigus-lgtm — CHANGES_REQUESTED** [review]

**Comment by intrigus-lgtm** [comment]
Thank you!

**Pull request merged**